interfaces/HO-Portal/src/app/components/update-fsp/update-fsp.component.ts

```diff
@@ -0,0 +1,160 @@
+
+  constructor(
+    private programsService: ProgramsServiceApiService,
+    private alertController: AlertController,
+    private translate: TranslateService,
+  ) {}
+
+  public startingAttributes: any[] = [];
+  public selectedFspAttributes: any[] = [];
+  public attributeDifference: any[] = [];
+  public startingFspName = '';
+  public selectedFspName = '';
+  public attributesToSave: {} = {};
+  public enableUpdateBtn = true;
+
+  ngOnInit() {
+    this.propertyModel = this.value;
+    this.startingFspName = this.value;
+    this.getFspAttributes(this.value);
+    this.startingAttributes = [...this.selectedFspAttributes];
+    this.enableUpdateBtn = false;
+  }
+
+  public updateChosenFsp() {
+    this.programsService
+      .updateChosenFsp(
+        this.referenceId,
+        this.selectedFspName,
+        this.attributesToSave,
+      )
+      .then(
+        () => {
+          this.inProgress = false;
+          this.actionResult(
+            this.translate.instant('common.update-success'),
+            true,
+          );
+        },
+        (error) => {
+          this.inProgress = false;
+          console.log('error: ', error);
+          if (error && error.error) {
+            this.actionResult(error.error.message);
+          }
+        },
+      );
+  }
+
+  public onFspChange({ detail }) {
+    this.getFspAttributes(detail.value);
+    if (!this.selectedFspAttributes.length) {
+      this.enableUpdateBtn = true;
+    }
+  }
+
+  public getFspAttributes(fspString: string) {
+    this.selectedFspAttributes = [];
+    this.attributesToSave = {};
+    this.selectedFspName = fspString;
+    if (this.fspList) {
+      const selectedFsp = this.fspList.find(
+        (fspItem) => fspItem.fsp === this.selectedFspName,
+      );
+
+      if (selectedFsp) {
+        this.selectedFspAttributes = selectedFsp.attributes;
+      }
+
+      this.attributeDifference = this.startingAttributes.filter(
+        (attr) => !this.selectedFspAttributes.includes(attr),
+      );
+    }
+  }
+
+  public onAttributeChange({ target, detail }) {
+    this.attributesToSave = {
+      ...this.attributesToSave,
+      [target.attributes['ng-reflect-name'].value]: detail.value.trim(),
+    };
+
+    this.checkAttributesFilled();
+  }
+
+  private checkAttributesFilled() {
+    if (
+      Object.values(this.attributesToSave).length <
+        this.selectedFspAttributes.length ||
+      Object.values(this.attributesToSave).includes('')
+    ) {
+      this.enableUpdateBtn = false;
+      return;
+    }
+
+    this.enableUpdateBtn = true;
+  }
+
+  private async actionResult(resultMessage: string, refresh: boolean = false) {
+    const alert = await this.alertController.create({
+      backdropDismiss: false,
+      message: resultMessage,
+      buttons: [
+        {
+          text: this.translate.instant('common.ok'),
+          handler: () => {
+            alert.dismiss(true);
+            if (refresh) {
+              window.location.reload();
```

Comment:
@elwinschmitz we could not come up with a better way to refresh. The pubsub-thing didn't seem to work (and we didn't know how it sould work).

The ideal situation is: that the popup stays open but gets refreshed, but that might be hard, so we went for the easy / second-best of refreshing the whole page.

Reply:
The idea is to _publish_ an event at this point, telling everybody else that "something has changed!"

So all the other components that _want_ to respond to that event can do/decide that themselves.

And in this case the "registrationDataChanged"-event is also applicable. (So very similar to a note being updated)